src/java.base/share/classes/java/util/stream/SliceOps.java

```diff
@@ -226,9 +228,16 @@ public static IntStream makeInt(AbstractPipeline<?, Integer, ?> upstream,
                                     long skip, long limit) {
         if (skip < 0)
             throw new IllegalArgumentException("Skip must be non-negative: " + skip);
+        long adjustedLimit = limit >= 0 ? limit : Long.MAX_VALUE;
 
         return new IntPipeline.StatefulOp<Integer>(upstream, StreamShape.INT_VALUE,
                                                    flags(limit)) {
+            @Override
+            long adjustSize(long size) {
+                return isParallel() ? super.adjustSize(size)
+                    : calcSize(super.adjustSize(size), skip, adjustedLimit);
```

Comment:
Do you think it is worth adding a comment that clarifies why serial and parallel cases are different?

Reply:
Ok